src/parsing/compileAstToExpression.ts

```diff
@@ -701,40 +745,42 @@ function instanceOf(
 	const expression = compile(astHelper.followPath(ast, ['argExpr', '*']), compilationOptions);
 	const sequenceType = astHelper.followPath(ast, ['sequenceType', '*']);
 	const occurrence = astHelper.followPath(ast, ['sequenceType', 'occurrenceIndicator']);
+	const type = astHelper.getAttribute(ast, 'type') as SequenceType;
 
 	return new InstanceOfOperator(
 		expression,
 		compile(sequenceType, disallowUpdating(compilationOptions)),
-		occurrence ? astHelper.getTextContent(occurrence) : ''
+		occurrence ? astHelper.getTextContent(occurrence) : '',
+		type
 	);
 }
 
 function integerConstantExpr(ast: IAST, _compilationOptions: CompilationOptions) {
-	return new Literal(
-		astHelper.getTextContent(astHelper.getFirstChild(ast, 'value')),
-		ValueType.XSINTEGER
-	);
+	return new Literal(astHelper.getTextContent(astHelper.getFirstChild(ast, 'value')), {
+		type: ValueType.XSINTEGER,
+		mult: SequenceMultiplicity.EXACTLY_ONE,
+	});
 }
 
 function stringConstantExpr(ast: IAST, _compilationOptions: CompilationOptions) {
-	return new Literal(
-		astHelper.getTextContent(astHelper.getFirstChild(ast, 'value')),
-		ValueType.XSSTRING
-	);
+	return new Literal(astHelper.getTextContent(astHelper.getFirstChild(ast, 'value')), {
+		type: ValueType.XSSTRING,
+		mult: SequenceMultiplicity.EXACTLY_ONE,
+	});
 }
 
 function decimalConstantExpr(ast: IAST, _compilationOptions: CompilationOptions) {
-	return new Literal(
-		astHelper.getTextContent(astHelper.getFirstChild(ast, 'value')),
-		ValueType.XSDECIMAL
-	);
+	return new Literal(astHelper.getTextContent(astHelper.getFirstChild(ast, 'value')), {
+		type: ValueType.XSDECIMAL,
+		mult: SequenceMultiplicity.EXACTLY_ONE,
+	});
 }
 
 function doubleConstantExpr(ast: IAST, _compilationOptions: CompilationOptions) {
-	return new Literal(
-		astHelper.getTextContent(astHelper.getFirstChild(ast, 'value')),
-		ValueType.XSDOUBLE
-	);
+	return new Literal(astHelper.getTextContent(astHelper.getFirstChild(ast, 'value')), {
+		type: ValueType.XSDOUBLE,
+		mult: SequenceMultiplicity.EXACTLY_ONE,
```

Comment:
All literals are EXACTLY_ONE, so you could opt to move this part to the Literal expression

Reply:
We tried to keep everything consistent and keep the coherence high, and since these things are done in compileAstToExpression as a general rule, kept it in there, even if it's not quite as terse as it could be that way.